app/src/main/java/rick_and_morty/ui/RickAndMortyActivity.kt

```diff
@@ -20,11 +21,23 @@ class RickAndMortyActivity : AppCompatActivity() {
     @Inject
     lateinit var eventBus: EventBus
 
+    @Inject
+    lateinit var realm: Realm
+
+    @Inject
+
+    lateinit var networkUtils: NetworkUtils
+
     private lateinit var navHostFragment: NavHostFragment
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        if (networkUtils.isInternetAvailable())
+        {
+            realm.executeTransactionAsync { it.deleteAll() }
+        }
```

Comment:
What if the API is not available? There would be no data you could show to the user.
You should first load the data and then update or delete the database.